pkg/tfbridge/provider_test.go

```diff
@@ -42,12 +43,333 @@ import (
+		// not processing the results of those with `schema.Resource` elements
+		// since it can get complicated. Nothing will get dropped and the
+		// validation error will pass through
+		{
+			name: "TypeList do not validate nested fields",
+			cloudVal: []interface{}{
+				map[string]interface{}{
+					"nestedProp":      "ABC",
+					"nestedOtherProp": "ABC",
+				},
+			},
+			schema: schema.Schema{
+				Type:     schema.TypeList,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"nested_prop": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ValidateDiagFunc: func(i interface{}, p cty.Path) diag.Diagnostics {
+								return diag.Errorf("Error")
+							},
+						},
+						"nested_other_prop": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ValidateDiagFunc: func(i interface{}, p cty.Path) diag.Diagnostics {
+								return nil
+							},
+						},
+					},
+				},
+			},
+			expectedProps: map[string]interface{}{
+				"collectionProp": []map[string]interface{}{
+					{
+						"nestedOtherProp": "ABC",
+						"nestedProp":      "ABC",
+					},
+				},
+			},
+			expectFailure: true,
+		},
+		{
+			name: "TypeSet Do not validate nested fields",
+			cloudVal: []interface{}{
+				map[string]interface{}{
+					"nestedProp": "ABC",
+				},
+			},
+			schema: schema.Schema{
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"nested_prop": {
+							Type:     schema.TypeString,
+							Required: true,
+							ValidateFunc: func(i interface{}, s string) ([]string, []error) {
+								return []string{}, []error{errors.New("Error")}
+							},
+						},
+					},
+				},
+			},
+			expectedProps: map[string]interface{}{
+				"collectionProps": []interface{}{
+					map[string]interface{}{
+						"nestedProp": "ABC",
+					},
+				},
+			},
+			expectFailure: true,
+		},
+	}
+	for _, tc := range tests {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			urn, err := resource.ParseURN("urn:pulumi:test::test::prov:index/test:Test::mainRes")
+			assert.NoError(t, err)
+			tfName := "prov_test"
+			sch := map[string]*schemav2.Schema{
+				"collection_prop": &tc.schema,
+				"other_prop": {
+					Type:     schema.TypeString,
+					Optional: true,
+				},
+			}
+			schemaMap := shimv2.NewSchemaMap(sch)
+			tfProvider := shimv2.NewProvider(&schema.Provider{
+				Schema: map[string]*schema.Schema{},
+				ResourcesMap: map[string]*schemav2.Resource{
+					"prov_test": {
+						Schema: sch,
+					},
+				},
+			})
+			inputs := resource.NewPropertyValue(tc.cloudVal)
+			plural := ""
+			if (tc.schema.Type == schema.TypeList || tc.schema.Type == schema.TypeSet) && tc.schema.MaxItems != 1 {
+				plural = "s"
+			}
+			inputsMap := resource.PropertyMap{
+				resource.PropertyKey("collectionProp" + plural): inputs,
```

Comment:
TerraformToPulumiNameV2 would be canonical. We trust it to turnaround vs PulumiToTerraformName.